### ISSUE TITLE
Tabs can jump around when selecting a tab because the editor actions grow/shrink

### ISSUE THREAD

**Issue by eamodio** [microsoft/vscode#115136]
Testing #114827

![tab-jumping](https://user-images.githubusercontent.com/641685/105896635-0fabd100-5fe5-11eb-9743-d38f7f3f8585.gif)

**Comment by bpasero** [comment]
@eamodio what would you suggest in this case? I think this is just a consequence of us having dynamic actions in the editor toolbar. The only way out I see is to not do that and keep editor actions in a toolbar in a different place?

**Comment by eamodio** [comment]
Yeah, not sure -- maybe something like reserve a certain amount of space for the actions so they can grow without wrapping the tabs as least until some threshold? I dunno.

Alternatively -- if bookmarks are visible, it would be nice to have the actions moved down to the "empty" right side.

**Comment by bpasero** [comment]
> if bookmarks are visible

Can you clarify, what are bookmarks in the context of VSCode?

> maybe something like reserve a certain amount of space for the actions so they can grow without wrapping the tabs as least until some threshold?

One idea I had was to simply never hide the actions when a group is inactive, but in fact this was not possible because actions are context key aware and some are only contributed when the group is active.

**Comment by eamodio** [comment]
Doh, I meant breadcrumbs

**Comment by bpasero** [comment]
Somehow I feel breadcrumbs belong more into the editor and are part of the editor than the tab and the editor actions, but it is an interesting idea.

**Comment by programmer247** [comment]
Is there some way to check how many actions will be shown for each file in the group? Then space could be reserved for the worst case in the group?

**Comment by complyue** [comment]
I've been annoyed by this behavior for quite a long time, routinely apply the trick https://github.com/microsoft/vscode/issues/46403#issuecomment-462935343 every restarts of the IDE.

Now I think maybe we can have a setting to request a fixed number of rows of tabs, then extraneous tabs go into a "..." drop down, as well as extraneous action-icons go into another "..." drop down. It should never "jump" again.

The fixed number of rows in each area, ideally can be ad-hoc manipulated by DnD gestures.

**Comment by BukeBeyond** [comment]
Tab wrap is a frequently used feature, and the implementation in VS Code is quite substandard.  There can not be any jumping around in the text editor position.  This happens even when the editor is focused and getting typed into, as source control/git scans the files in the background, it adds and removes "Editor Actions", the tabs, the editor position jump around and have a mind of their own...  

These "Editor Actions" need to have their own area or row.  They belong to a single file, not to a collection of files and tabs.  So it is a confusing design that they share that space anyway, besides the unstable positioning and interference with usability.

Sharing the right side of the breadcrumb area is not a bad idea, as the breadcrumb belongs to a single file as well.  Being able to turn them off completely, or always show the fixed ... menu should be another option.

**Comment by carlfriedrich** [comment]
Totally agree with @BukeBeyond. There is a frequently visted [thread on StackOverflow](https://stackoverflow.com/q/47266613/3018229) on this topic which shows that it is bugging quite a lot of users.

I managed to [move the action buttons to the breadcrumb bar](https://stackoverflow.com/a/69788835/3018229) using the [Customize UI](https://marketplace.visualstudio.com/items?itemName=iocave.customize-ui) extension and can confirm that this results in a way better user experience:

![Action buttons in breadcrumb bar](https://i.stack.imgur.com/dMCE4.png)

**Comment by florom** [comment]
Hi Devs,

is there an Update on this one? 
I started using VS Code and I find it really annoying as mentioned above a few times.
So I will definitly watch this issue from now on. Hopefully it gets more attention.

**Comment by bpasero** [comment]
At least for the editor actions you can decide to show them in the title bar or hide them which will avoid the issue of jumping wrapped tabs at least for when actions change:

![image](https://github.com/microsoft/vscode/assets/900690/9c78e38f-a7a6-4387-82bc-59b6547f12ef)

This feature can be tried out in our insiders build: You can give our preview releases a try from: https://code.visualstudio.com/insiders/

//cc @benibenj for enabling this capability

**Comment by a18rhodes** [comment]
Any update here? This has been open for almost 3 years now.

I realize there is a workaround, but I do not want to have install the insiders build just to achieve this, and it seems that the extension workaround may require re-enabling the extension every time VSCode is relaunched - both of which are cumbersome.

**Comment by bpasero** [comment]
Everything we have in insiders gets into stable on the next release, so you should be good to go with the workaround at least.

**Comment by a18rhodes** [comment]
@bpasero, thanks for the info! The workaround is working for me.

**Comment by fr-an-k** [comment]
This is still an issue in the latest version (1.94.2)
